students/aravichander/lesson09/Mailroom_class.py

```diff
@@ -3,68 +3,105 @@
+		self._donations = [] if donations is None else donations 
 
+	def add_donation(self,donation):
+		self._donations.append(donation)
 
 	@property
-	def numberofdonors(self)
+	def name(self):
+		return self._name
 
 	@property
-	def totaldonationsofdonors(self)
-
-	@classmethod
-	def memberdonations(cls..etc.)
-
-
-
-
-
-
-
-
-
-
-
-
-Donors = {}
-
-def createlist():
-	Donors["Bill Gates"] = [10000,12000]
-	Donors["Jeff Bezos"]=[50]
-	Donors["Mark Zuckerberg"] = [500,600,700]
-	Donors["Paul Allen"] = [250,350,450]
-	Donors["King of Siam"] = [200,250]
-
-def getlist():
-	return Donors 
-	
-
-def printdonors():
-	for x,v in Donors.items():
-		print("\n{} donated ${}".format(x,v))
+	def donations(self):
+		return self._donations
 
-def thankall():
-	for x in Donors:
-		totaldonated = sum(Donors[x])
-		letterofthanks(x,totaldonated)
+	@property
+	def sum_donation(self):
+		return sum(self._donations)
 
-def letterofthanks(donorname,totaldonated):
+	@property
+	def number_of_donations(self):
+		return len(self._donations)
+#My ipython was going nuts so I had to manually calculate the average >=(
+	@property
+	def donation_average(self):
+		return sum(self._donations)/len(self._donations)
+
+
+
+class Donor_db:
+	def __init__(self):
+		self._donors = {}
+
+	def add_donor(self,donor):
+		self._donors[donor.name.lower()] = donor
+
+	# def all_donors_donations(self): Couldn't figure out the issue here re:testing
+	# 	total_sum = 0
+	# 	for name,donor_obj in self._donors.items():
+	# 		print(name)
+	# 		total_sum = total_sum + donor_obj.sum_donation
+	# 	return total_sum
+
+	def get_donor(self, donor_name):
+	 	return self._donors[donor_name.lower()]
+
+	def print_Db(self):
+		for name,donor_obj in self._donors.items(): 
+			print("\n{} donated individual donations of ${} for a total of ${}".format(name,donor_obj.donations,donor_obj.sum_donation))
+
+	def get_sorted_donors(self): 
+		return sorted(self._donors)
+
+	def num_donors(self):
+		return len(self._donors)
+
+	# @property
+	# def number_of_donors(self):
+	# 	return len(self._donors)
+
+#################################################################################
+
+
+def create_db():
+	a = Donor("Bill Gates")
+	a.add_donation(10000)
+	a.add_donation(12000)
+	b = Donor("Jeff Bezos")
+	b.add_donation(50)
+	c = Donor("Mark Zuckerberg")
+	c.add_donation(500)
+	c.add_donation(600)
+	c.add_donation(700)
+	d = Donor("Paul Allen")
+	d.add_donation(250)
+	d.add_donation(350)
+	d.add_donation(450)
+	e = Donor("King of Siam")
+	e.add_donation(200)
+	e.add_donation(250)
+	donor_db = Donor_db()
+	donor_db.add_donor(a)
+	donor_db.add_donor(b)
+	donor_db.add_donor(c)
+	donor_db.add_donor(d)
+	donor_db.add_donor(e)
+	return donor_db
+
+def thankall(db):
+	for name in db.get_sorted_donors():
+		donor = db.get_donor(name)
+		totaldonated = donor.sum_donation
+		letterofthanks(name,totaldonated)
+
+def letterofthanks(donorname,totaldonated): #Potentially create method for letterofthanks within donor class
```

Comment:
if you make `letter_of_thanks()` a method of the Donor class, then it would have all the information it needed (in self)